internal/templates/commands/render.go

```diff
@@ -0,0 +1,176 @@
+			`"http://example.com/myremotetarball.tgz" os for a non-Git template in a remote tarball.`,
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "spec",
+		Example: "path/to/spec.yaml",
+		Target:  &r.fv.spec,
+		Default: "./spec.yaml",
+		Usage:   "The path of the .yaml file within the unpacked template directory that specifies how the template is rendered.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "dest",
+		Aliases: []string{"d"},
+		Example: "/my/git/dir",
+		Target:  &r.fv.dest,
+		Default: ".",
+		Usage:   "Required. The target directory in which to write the output files.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "git-protocol",
+		Example: "https",
+		Default: "https",
+		Target:  &r.fv.gitProtocol,
+		Usage:   "Either ssh or https, the protocol for connecting to GitHub. Only used if the template source is GitHub.",
+	})
+	sect.StringMapVar(&cli.StringMapVar{
+		Name:    "input",
+		Example: "foo=bar",
+		Target:  &r.fv.inputs,
+		Usage:   "The key=val pairs of template values; may be repeated.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "log-level",
+		Example: "info",
+		Default: "warning",
+		Target:  &r.fv.logLevel,
+		Usage:   "How verbose to log; any of debug|info|warning|error.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "force-overwrite",
+		Target:  &r.fv.forceOverwrite,
+		Default: false,
+		Usage:   "If an output file already exists in the destination, overwrite it instead of failing.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "keep-temp-dirs",
+		Target:  &r.fv.keepTempDirs,
+		Default: false,
+		Usage:   "Preserve the temp directories instead of deleting them normally.",
+	})
+
+	return set
+}
+
+func (r *Render) parseFlags(args []string) error {
+	if err := r.Flags().Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	if r.fv.template == "" {
+		return fmt.Errorf("-template is required")
+	}
+
+	return nil
+}
+
+func (r *Render) Run(ctx context.Context, args []string) error {
+	if err := r.parseFlags(args); err != nil {
+		return err
+	}
+
+	if r.fs == nil { // allow filesystem interaction to be faked for testing
+		r.fs = &realFS{}
+	}
+
+	if err := destOK(r.fs, &r.fv); err != nil {
+		return err
+	}
+
+	return fmt.Errorf("stub")
+}
+
+// destOK makes sure that the output directory looks sane; we don't want to clobber the user's
+// home directory or something.
+func destOK(fs fs, fv *flagValues) error {
+	fi, err := fs.Stat(fv.dest)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("the destination directory doesn't exist: %q", fv.dest)
+		}
+		return fmt.Errorf("os.Stat(%s): %w", fv.dest, err)
+	}
+
+	if !fi.IsDir() {
+		return fmt.Errorf("the destination %q is not a directory", fv.dest)
+	}
+
+	return nil
+}
+
+// An interface that allows for a fake filesystem to be used in tests.
+type fs interface {
+	Stat(string) (os.FileInfo, error)
+}
```

Comment:
I think you might want to use the stdlib fs interfaces https://pkg.go.dev/io/fs

Reply:
Done.